librecad/src/main/lc_application.cpp

```diff
@@ -0,0 +1,24 @@
+#include <QStringList>
+#include <QFileOpenEvent>
+
+#include "lc_application.h"
+
+LC_Application::LC_Application(int &argc, char **argv)
+    : QApplication(argc, argv)
+{
+    QStringList files = QStringList();
+}
+
+// This is only used until the event filter is in place in mainwindow
+bool LC_Application::event(QEvent *event)
+{
+    if (event->type() == QEvent::FileOpen) {
```

Comment:
This event is OSX specific, it is not fired on Windows or Linux, so I couldn't test this yet.
Even when `files` is populated here, `filesList()` is never called after the event loop starts.
Seems to be a WOM (write only memory) :wink:
I haven't got yet, what OSX want to signal with `QEvent::FileOpen`. On Windows and Linux, when I double click a DXF file, a new instance of LibreCAD opens with this file, as the filename is pushed into `main()` and handled by `handleArgs()` there.

Reply:
anything that is opened by double clicking the file in finder unfortunately doesn't get pushed into main but does get sent as a fileopenevent, unfortunately one can't make it wait until the eventfilter is installed, as such the LC_Application is needed to catch any early events and then main.cpp uses that to send to the window.

any subsequent fileopenevents are handled by the eventfilter